src/tests/ftest/tags.py

```diff
@@ -650,84 +726,206 @@ def print_verbose(*args):
+                    yaml_data = TestYamlData(file_path, method_name)
+                    for key in key_list:
+                        output[-1].append(yaml_data.value(key))
+                except Exception as error:      # pylint: disable=broad-except
+                    print(f"<< Error processing yaml keys {key_list} for {short_file_path} >>")
+                    raise error
+
+    if csv_file:
+        with open(csv_file, 'w', newline='\n') as csvfile:
+            csv_writer = csv.writer(csvfile)
+            for line in output:
+                csv_writer.writerow(line)
+        print(f"Generated {csv_file} with {len(output)} records")
+
+    else:
+        width = []
+        for line in output:
+            for index, column in enumerate(line):
+                size = len(str(column))
+                if index == len(width):
+                    width.append(size)
+                elif size > width[index]:
+                    width[index] = size
+        format_line = "  ".join(list(map(lambda x: f"{{:{x}}}", width)))
+        for line in output:
+            print(format_line.format(*list(map(lambda x: str(x), line))))
+
+    return 0 if output else 1
+
+
+def __comma_separated_arg(val):
+    """Parse a comma-separated argument.
+
+    Args:
+        val (str): string to parse comma-separated values from
+
+    Returns:
+        set: comma-separated strings converted to a set
 
     Raises:
         ArgumentTypeError: if val is invalid
     """
     if not val:
-        raise ArgumentTypeError("tags cannot be empty")
+        raise ArgumentTypeError("comma-separated argument cannot be empty")
     try:
         return set(map(str.strip, val.split(",")))
     except Exception as err:  # pylint: disable=broad-except
-        raise ArgumentTypeError(f"Invalid tags: {val}") from err
+        raise ArgumentTypeError(f"Invalid comma-separated argument: {val}") from err
 
 
 def main():
-    """main function execution"""
-    description = '\n'.join([
-        'Commands',
-        '  lint - lint ftest avocado tags',
-        '  list - list ftest avocado tags associated with test files',
-        '  dump - dump the file/class/method/tag structure for test files',
-        '  unit - run self unit tests'
-    ])
-
-    parser = ArgumentParser(formatter_class=RawDescriptionHelpFormatter, description=description)
-    parser.add_argument(
-        "command",
-        choices=("lint", "list", "dump", "unit"),
-        help="command to run")
-    parser.add_argument(
+    """Main function execution.
+
+    Returns:
+        int: 0 = success
+             1 = error
+             2 = code problem
+    """
+    parser = ArgumentParser(prog='tags')
+    subparsers = parser.add_subparsers(
+        title='options for the tags command',
+        dest='command')
+
+    # Parser for the "data" command and its optional arguments
+    data_parser = subparsers.add_parser(
+        'data',
```

Comment:
I think something like "config" or "parse-config", "dump-config" etc. is more clear than "data", which is generic.

Reply:
Updated.